lib/sacastats_web/controllers/character_controller.ex

```diff
@@ -17,6 +18,42 @@ defmodule SacaStatsWeb.CharacterController do
     redirect(conn, to: conn.request_path <> "/general")
   end
 
+  def latest_session(%Plug.Conn{} = conn, %{"character_name" => name}) do
+    case Characters.get_by_name(name) do
+      {:ok, %Character{} = char} ->
+        case Session.get_latest_timestamp(char.name_first_lower) do
```

Comment:
Since you have access to the character_id in `char`, do you think you could change `get_latest_timestamp/1` to take a character ID instead so it doesn't have to call `get_by_name/1` again?

Non-blocking: these nested `case`s could probably be converted to a `with` statement

Reply:
Good catch! How would you use the `with` statement in this way when the second case is dependent on the first's outcome?

Reply:
It would look something like this (untested of course):
```elixir
with {:ok, %Character{} = char} <- Characters.get_by_name(name),
      %Events.PlayerLogin{timestamp: timestamp} <- Session.get_latest_timestamp(char.name_first_lower) do
  redirect(...)
else
  # handle the `nil`, `:error`, and `:not_found` cases here
end
```